src/components/ScriptExplorer/SignatureImporter.jsx

```diff
@@ -304,72 +314,159 @@ class SignatureImporter extends React.Component {
       errback("Not enough signatures (must be exactly one for each input).");
       return;
     }
-    if (inputsSignatures.length > inputs.length) {
-      errback("Too many signatures (must be exactly one for each input).");
-      return;
+
+    if (inputsSignatures.length % inputs.length !== 0) {
+      errback(
+        "Not the right number of signatures (must be exactly one for each input)."
+      );
     }
 
+    const numSignatureSets = inputsSignatures.length / inputs.length;
```

Comment:
`numSignatureSets` cannot be larger than number of required signers (also == to the number of signature importers)